git-branchless/src/commands/move.rs

```diff
@@ -168,16 +217,163 @@ pub fn r#move(
+                // component.
+                let nearest_dest_index = possible_destinations.len() - 1;
+                if possible_destinations.len() > 1
+                    && !dag.query().is_ancestor(
+                        possible_destinations[nearest_dest_index - 1].into(),
+                        possible_destinations[nearest_dest_index].into(),
+                    )?
+                {
+                    writeln!(
+                        effects.get_output_stream(),
+                        "This operation cannot be completed because the {} at {}\n\
+                        has multiple possible parents also being moved. Please retry this operation\n\
+                        without this {}, or with only 1 possible parent.",
+                        if component.count()? == 1 {
+                            "commit"
+                        } else {
+                            "range of commits rooted"
+                        },
+                        component_root,
+                        if component.count()? == 1 {
+                            "commit"
+                        } else {
+                            "range of commits"
+                        },
+                    )?;
+                    return Ok(ExitCode(1));
+                }
+
+                let nearest_component =
+                    exact_components.get(&possible_destinations[nearest_dest_index]);
+                if let Some(nearest_component) = nearest_component {
+                    // The current component could be descended from any commit
+                    // in nearest_component, not just it's head.
+                    let dest_ancestor = dag
+                        .query()
+                        .ancestors(CommitSet::from(component_head))?
+                        .intersection(nearest_component);
+                    let head = dag.query().heads(dest_ancestor.clone())?.first()?;
+                    if let Some(head) = head {
+                        NonZeroOid::try_from(head)?
+                    } else {
+                        dest_oid
+                    }
+                } else {
+                    // TODO this should be unreachable
+                    dest_oid
+                }
+            };
+
+            let component_parent = match dag.get_only_parent_oid(*component_root) {
+                Ok(oid) => oid,
+                Err(_) => {
+                    writeln!(
+                        effects.get_output_stream(),
+                        "The --exact flag can only be used to move ranges or commits with exactly 1 parent.",
+                    )?;
+                    return Ok(ExitCode(1));
+                }
+            };
+
+            let component_children: CommitSet = dag
+                .query()
+                .children(component.clone())?
+                .difference(&dag.obsolete_commits)
+                .difference(component);
```

Comment:
```suggestion
                .difference(component)
                .difference(&dag.obsolete_commits);
```

It probably doesn't matter too much in this case, but let's do this in the opposite order. Generally, we want to operate on the smallest sets first for performance.

Reply:
Ah, OK, thank you. That makes sense!